src/util-path.c

```diff
@@ -66,6 +66,40 @@ int PathIsRelative(const char *path)
     return PathIsAbsolute(path) ? 0 : 1;
 }
 
+/**
+ * \brief Wrapper to join a directory and filename and resolve using realpath
+ *   _fullpath is used for WIN32
+ * \param out_buf output buffer.  Up to PATH_MAX will be written.  Unchanged on exit failure.
+ * \param buf_len length of output buffer
+ * \param dir the directory
+ * \param fname the filename
+ * \retval TM_ECODE_OK on success
+ * \retval TM_ECODE_FAILED on failure
+ */
+TmEcode PathJoin (char *out_buf, uint16_t buf_len, const char *const dir, const char *const fname)
+{
+#ifdef OS_WIN32
+#define DIRECTORY_SEPARATOR '\\'
+#else
+#define DIRECTORY_SEPARATOR '/'
+#endif
+    uint16_t max_path_len = (buf_len > PATH_MAX) ? PATH_MAX : buf_len;
+    int bytes_written = snprintf(out_buf, max_path_len, "%s%c%s", dir, DIRECTORY_SEPARATOR, fname);
+    if (bytes_written <= 0) {
+        SCLogError(SC_ERR_SPRINTF, "Could not join filename to path");
+        SCReturnInt(TM_ECODE_FAILED);
```

Comment:
Please add `SCEnter()`  to this function.

Reply:
done.